RecoHI/Configuration/python/customise_PPwithHI.py

```diff
@@ -0,0 +1,506 @@
+        nSigmaZ = 4.0,
+        sigmaZVertex = 4.0,
+        fixedError = 5.0,#a fourth the size of the pp version
+        VertexCollection = "firstStepPrimaryVertices",
+        ptMin = 2.0,
+        useFoundVertices = True,
+        originRadius = 3.5
+))
+    process.tobTecStepTrackingRegionsPair = _globalTrackingRegionWithVertices.clone(RegionPSet=dict(
+        precise = True,
+        useMultipleScattering = False,
+        useFakeVertices       = False,
+        beamSpot = "offlineBeamSpot",
+        useFixedError = True,#this means use fixedErrorBelow
+        nSigmaZ = 4.0,
+        sigmaZVertex = 4.0,
+        fixedError = 7.5,#a fourth the size of the pp version
+        VertexCollection = "firstStepPrimaryVertices",
+        ptMin = 2.0,
+        useFoundVertices = True,
+        originRadius = 6.0
+))
+    process.tobTecStepTrajectoryFilter.minPt = 2.0
+
+    return process
+
+#copied almost exactly from RecoTracker/Configuration/python/customiseClusterCheckForHighPileup.py
+#some threshholds have been retuned however
+#also need to reenable doClusterCheck because it is turned off by default in phase 1
+def customiseClusterCheck(process):
+    _maxPixel = 100000
+
+    #this cut was tuned with XeXe MC: The old cut used for peripheral PbPb in 2015 is listed in comment below    
+    #_cut = "strip < 1000000 && pixel < 100000 && (strip < 50000 + 10*pixel) && (pixel < 5000 + strip/7.)"
+    _cut = "strip < 1000000 && pixel < 100000 && (strip < 50000 + 10*pixel) && (pixel < 5000 + strip/2.)"
+
+    _maxElement = 1000000
+
+    for module in process._Process__producers.values():
+        cppType = module._TypedParameterizable__type
+
+        # cluster multiplicity check
+        if cppType == "ClusterCheckerEDProducer":
+            module.doClusterCheck = True #added this line (not in pp config)
+            module.MaxNumberOfPixelClusters = _maxPixel
+            module.cut = _cut
+        if hasattr(module, "ClusterCheckPSet"):
+            module.ClusterCheckPSet.MaxNumberOfPixelClusters = _maxPixel
+            module.ClusterCheckPSet.doClusterCheck = True #added this line (not in pp config)
+            # PhotonConversionTrajectorySeedProducerFromQuadruplets does not have "cut"...
+            if hasattr(module.ClusterCheckPSet, "cut"):
+                module.ClusterCheckPSet.cut = _cut
+
+
+        if cppType in ["PixelTripletLargeTipEDProducer", "MultiHitFromChi2EDProducer"]:
+            module.maxElement = _maxElement
+        if hasattr(module, "OrderedHitsFactoryPSet") and hasattr(module.OrderedHitsFactoryPSet, "GeneratorPSet"):
+            #next line is in pp config but we comment it to be safe by changing more modules...
+            #if module.OrderedHitsFactoryPSet.GeneratorPSet.ComponentName.value() in ["PixelTripletLargeTipGenerator", "MultiHitGeneratorFromChi2"]:
+            module.OrderedHitsFactoryPSet.GeneratorPSet.maxElement = _maxElement
+    
+    return process
+
+def customisePPwithHI(process):
```

Comment:
I thought we had this kind of modification in the past.
Did it get lost in past releases?

Reply:
We did have in past releases for a reprocessing of peripheral PbPb with pp reco (see 75X).  This customization is a bit more evolved with respect to that one.   I guess you could say it was lost, since it's no longer present in the master and was never adapted for Phase I.  I'm not sure we would use the same exact customization for the two tasks, but it would likely be similar.

Reply:
Looking back, the solution that was a customization in 75X, became an era pA_2016.
Please check what's inside it and if it is missing things, define a pp_on_XeXe_2017 and put all the modifications in it.
After that, the T0 data processing can pick it up correctly.

I still need to figure out how to apply the repacking calls for strip VR

Reply:
if pA_2016  is a relatively strict subset of pp_on_XeXe_2017, please use that fact

Reply:
Ah, right.  Ok, I'm on it.  I will add a corresponding relval wf as well, if you don't object.